src/sentry/models/projectownership.py

```diff
@@ -116,51 +116,71 @@ def get_owners(cls, project_id, data):
 
-        Returns a tuple of (auto_assignment_enabled, list_of_owners).
+        Returns a tuple of (auto_assignment_enabled, list_of_owners, assigned_by_codeowners: boolean).
         """
         from sentry.models import ProjectCodeOwners
 
         with metrics.timer("projectownership.get_autoassign_owners"):
             ownership = cls.get_ownership_cached(project_id)
             codeowners = ProjectCodeOwners.get_codeowners_cached(project_id)
-
+            assigned_by_codeowners = False
             if not (ownership or codeowners):
-                return False, []
+                return False, [], assigned_by_codeowners
 
             if not ownership:
                 ownership = cls(project_id=project_id)
 
-            ownership.schema = cls.get_combined_schema(ownership, codeowners)
+            ownership_rules = cls._matching_ownership_rules(ownership, project_id, data)
+            codeowners_rules = (
+                cls._matching_ownership_rules(codeowners, project_id, data) if codeowners else []
+            )
 
-            rules = cls._matching_ownership_rules(ownership, project_id, data)
-            if not rules:
-                return ownership.auto_assignment, []
+            if not (codeowners_rules or ownership_rules):
+                return ownership.auto_assignment, [], assigned_by_codeowners
 
-            # We want the last matching rule to take the most precedence.
-            owners = [owner for rule in rules for owner in rule.owners]
-            owners.reverse()
-            actors = {
-                key: val
-                for key, val in resolve_actors({owner for owner in owners}, project_id).items()
-                if val
-            }
-            actors = [actors[owner] for owner in owners if owner in actors][:limit]
+            ownership_actors = cls._find_actors(project_id, ownership_rules, limit)
+            codeowners_actors = cls._find_actors(project_id, codeowners_rules, limit)
 
             # Can happen if the ownership rule references a user/team that no longer
             # is assigned to the project or has been removed from the org.
-            if not actors:
-                return ownership.auto_assignment, []
+            if not (ownership_actors or codeowners_actors):
+                return ownership.auto_assignment, [], assigned_by_codeowners
+
+            # Ownership rules take precedence over codeowner rules.
+            actors = [*ownership_actors]
+            if len(actors) < limit and codeowners_actors:
+                actors = [*ownership_actors, *codeowners_actors][:limit]
+                assigned_by_codeowners = True
 
             from sentry.models import ActorTuple
 
-            return ownership.auto_assignment, ActorTuple.resolve_many(actors)
+            return (
+                ownership.auto_assignment,
+                ActorTuple.resolve_many(actors),
+                assigned_by_codeowners,
```

Comment:
I'm probably going to take another look at this PR but I kind of don't love the idea of updating the return value here just for analytics. But I do see that doing this way means we can record the group.

I'm wondering if for now it would be enough just to have the analytics record when we know that
`ownership.auto_assignment = True` and `ActorTuple.resolve_many(actors)` is not empty. 

so I'm thinking you could keep the `assigned_by_codeowners = False` at the top but then the return statements could say as they are and the at the end we could do something like

```python 
if len(actors) < limit and codeowners_actors:
    actors = [*ownership_actors, *codeowners_actors][:limit]
    assigned_by_codeowners = True

    from sentry.models import ActorTuple
    
    owners = ActorTuple.resolve_many(actors)
    if assigned_by_codeowners and ownership.auto_assignment and owners:
        analytics.record(
            "codeowners.assignment",
             codeowners_id=codeowners.id,
             project_id=project.id,    
        )
return ownership.auto_assignment, owners
```

Reply:
@MeredithAnya hmm I think it's ok to keep it where it is bc we need to record the group id for the questions we want answered.